daemon/src/routes_maker.rs

```diff
@@ -208,6 +208,63 @@ pub fn index<'r>(_paths: PathBuf, _auth: Authenticated) -> impl Responder<'r, 's
     Ok::<(ContentType, Cow<[u8]>), Status>((ContentType::HTML, asset.data))
 }
 
+#[derive(Debug, Clone, Deserialize)]
+pub struct WithdrawRequest {
+    pub address: bdk::bitcoin::Address,
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_btc")]
+    pub amount: bdk::bitcoin::Amount,
+    pub fee: f32,
+}
+
+#[rocket::post("/withdraw", data = "<withdraw_request>")]
+pub async fn post_withdraw_request(
+    withdraw_request: Json<WithdrawRequest>,
+    wallet: &State<Address<wallet::Actor>>,
+    network: &State<Network>,
+    _auth: Authenticated,
+) -> Result<status::Accepted<String>, HttpApiProblem> {
+    tracing::info!(
+        "Withdrawing {:?} to {} with fee {}",
+        withdraw_request.amount,
+        withdraw_request.address,
+        withdraw_request.fee
+    );
+
+    let amount = if withdraw_request.amount == bdk::bitcoin::Amount::ZERO {
+        None
+    } else {
+        Some(withdraw_request.amount)
+    };
+
+    let txid = wallet
+        .send(wallet::Withdraw {
+            amount,
+            address: withdraw_request.address.clone(),
+            fee: Some(bdk::FeeRate::from_sat_per_vb(withdraw_request.fee)),
+        })
+        .await
+        .map_err(|e| {
+            HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
+                .title("Withdraw failed")
+                .detail(e.to_string())
+        })?
+        .map_err(|e| {
+            HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
+                .title("Withdraw failed")
+                .detail(e.to_string())
+        })?;
+
+    tracing::info!(%txid, "Withdraw successful");
+    let url = match network.inner() {
+        Network::Bitcoin => format!("https://mempool.space/tx/{}", txid),
+        Network::Testnet => format!("https://mempool.space/testnet/tx/{}", txid),
+        Network::Signet => format!("https://mempool.space/signet/tx/{}", txid),
+        Network::Regtest => txid.to_string(),
+    };
+
+    Ok(status::Accepted(Some(url)))
```

Comment:
Is the use of `Accepted` here intentional?

With `Accepted` you could include a `Location` header in the response which points the user to a "progress" page about the status of the processing.

I guess one could spin it such that the mempool page is about the progress of getting the TX included? I think I wouldn't bother with these semantics though and just return 200 OK.

Reply:
See https://httpstatuses.com/202.

Reply:
It wasn't on purpose but I learned something new :) 
Changing to 200 OK instead